src/qt/listctrl.cpp

```diff
@@ -230,7 +230,7 @@ class wxQtListModel : public QAbstractTableModel
                     : QVariant();
 
             case Qt::TextAlignmentRole:
-                return columnItem.m_align;
+                return QVariant((int)columnItem.m_align);
```

Comment:
Sorry, what does this fix? It seems non-obvious why is this better than relying on an implicit conversion to `QVariant`, so a comment explaining this would be useful.

Reply:
Because the type has changed, without this patch, there's warnings with default C++ version:

```
...\src\qt\listctrl.cpp(233): warning C4927: illegal conversion; more than one user-defined conversion has been implicitly applied
  ...\src\qt\listctrl.cpp(233): note: while calling the constructor 'QVariant::QVariant(int)'
  ...\Qt\5.15.2\msvc2019_64\include\QtCore/qvariant.h(219): note: see declaration of 'QVariant::QVariant'
```

on C++20 there's errors:

```
...\src\qt\listctrl.cpp(233): error C2440: 'return': cannot convert from 'const Qt::Alignment' to 'QVariant'
  ...\src\qt\listctrl.cpp(233): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
```

I just updated it to use `static_cast<int>` instead.

Reply:
Thanks for the explanation! Ideal would be to use `std::to_underlying` as it's more readable but it's only in C++23... Maybe we should provide a fallback definition for it in some private header for the future.